request-manage-service/controller/RequestManage.controller.js

```diff
@@ -0,0 +1,369 @@
+//  .then(data => {
+//    const fareCalc = req.body.quantity * data.price;       // check field name in vendor management service
+//
+//    const manageRequest = {
+//        userId: req.headers['x-userid'],
+//        vendorId: req.body.vendorId,
+//        categoryId : req.body.categoryId,
+//        quantity: req.body.quantity,
+//        fare: fareCalc,
+//        status: "Pending",
+//    };
+//
+//      let checkValid = v.validate(manageRequest, manageRequestSchema)
+//      if (checkValid.valid) {
+//        // Save in the database
+//        ManageRequest.create(manageRequest)
+//          .then((data) => {
+//            data = appendHeaders(req, data)
+//            res.status(201).send(data);
+//          })
+//          .catch((err) => {
+//            if (err.name === 'SequelizeUniqueConstraintError') {
+//              res.status(409).send({
+//                message: "Duplicate request placed."
+//              })
+//              return
+//            }
+//            res.status(500).send({
+//              message:
+//                err.message || "Some error occurred while creating the request.",
+//            });
+//          });
+//      } else {
+//        console.log(checkValid)
+//        res.status(400).send({
+//          message:
+//            checkValid.errors
+//        });
+//        return
+//      }
+//
+//  })
+//  .catch(error => {
+//    console.error('Error while fetching details from vendor management service', error);
+//  });
+//};
+
+
+
+//Create request in database     -- local
+exports.createRequest = (req,res) => {
+
+    const fareCalc = 67.90;       // check field name in vendor management service
+
+    const manageRequest = {
+        userId: req.body.userId,
+        vendorId: req.body.vendorId,
+        categoryId : req.body.categoryId,
+        quantity: req.body.quantity,
+        fare: fareCalc,
+        status: "Pending",
+    };
+
+      let checkValid = v.validate(manageRequest, manageRequestSchema)
+      if (checkValid.valid) {
+        // Save in the database
+        ManageRequest.create(manageRequest)
+          .then((data) => {
+            data = appendHeaders(req, data)
+            res.status(201).send(data);
+          })
+          .catch((err) => {
+            if (err.name === 'SequelizeUniqueConstraintError') {
+              res.status(409).send({
+                message: "Duplicate request placed."
+              })
+              return
+            }
+            res.status(500).send({
+              message:
+                err.message || "Some error occurred while creating the request.",
+            });
+          });
+      } else {
+        console.log(checkValid)
+        res.status(400).send({
+          message:
+            checkValid.errors
+        });
+        return
+      }
+};
+
+
+// Find all request for given user  -- local
+exports.getAllRequest = (req, res) => {
+  const userId = req.headers['x-userid'];
+  const role = "CONSUMER";
```

Comment:
I believe later this role will be fetched from user management service.
Instead of doing this for every request I would recommend having a separate api for Recycler and Customer endpoints for eg:- 
If User is navigating then from app they will hit  /usermanagement/user api
If Recycler is navigating then from app they will hit  /usermanagement/recycler api

Reply:
This would remove dependency from user microservice.

Reply:
Will make this change